adam-core/src/main/scala/org/bdgenomics/adam/converters/FastqRecordConverter.scala

```diff
@@ -41,6 +41,114 @@ import scala.collection.JavaConversions._
 private[adam] class FastqRecordConverter extends Serializable with Logging {
 
   /**
+   * Parse 4 lines at a time
+   * @see parseReadPairInFastq
+   * *
+   */
+  private[converters] def readNameSuffixAndIndexOfPairMustMatch(readName: String,
+                                                                isFirstOfPair: Boolean): Unit = {
+    val firstReadSuffix = """[/ +_]1$""".r
+    val secondReadSuffix = """[/ +_]2$""".r
+
+    val isSecondOfPair = !isFirstOfPair
+
+    val match1 = firstReadSuffix.findAllIn(readName)
+    val match2 = secondReadSuffix.findAllIn(readName)
+
+    if (match1.nonEmpty && isSecondOfPair)
+      throw new IllegalArgumentException(
```

Comment:
These exceptions are thrown without considering `ValidationStringency`.  If the stringency is lenient or silent, is it possible to continue processing?